test/parallel/test-child-process-urlfile.mjs

```diff
@@ -0,0 +1,42 @@
+import { isWindows, mustCall, mustSucceed, mustNotCall } from '../common/index.mjs';
+import { strictEqual } from 'node:assert';
+import { pathToFileURL } from 'node:url';
+import cp from 'node:child_process';
+import tmpdir from '../common/tmpdir.js';
+
+tmpdir.refresh();
+
+const cwd = tmpdir.path;
+const whichCommand = isWindows ? 'where.exe cmd.exe' : 'which pwd';
+const pwdFullPath = `${cp.execSync(whichCommand)}`.trim();
+const pwdUrl = pathToFileURL(pwdFullPath);
+const pwdCommandAndOptions = isWindows ?
+  [pwdUrl, ['/d', '/c', 'cd'], { cwd: pathToFileURL(cwd) }] :
```

Comment:
Can you also add a test where javascript object representation of the URL is used?